### PR TITLE
frontend/basicReviewsChart

### PR DESCRIPTION
**Pull request by zmattingly**
Positive Reviews chart is now in place.

screen shot 2018-06-12 at 3 07 52 pm

Trello Ticket: https://trello.com/c/fohy2pRz

### CODE REVIEW

**Comment by zmattingly** [comment]
Heyo @John-Michael-Long, can you give this a review?

File: client/src/Components/ReviewsChart.jsx [hunk 1]
```diff
@@ -8,20 +9,104 @@ import './../styles/ReviewsChart.scss';
 //   |____|_  /\___  >\_/ |__|\___  >\/\_//____  >\______  /___|  (____  /__|   |__|  
 //          \/     \/             \/           \/        \/     \/     \/             
 
-const ReviewsChart = (props) => {
```
Comment (John-Michael-Long) [inline client/src/Components/ReviewsChart.jsx:11]: I believe Airbnb dictates that a single argument doesn't need parens

File: client/src/Components/ReviewsChart.jsx [hunk 2]
```diff
@@ -8,20 +9,104 @@ import './../styles/ReviewsChart.scss';
...
+    if (this.state.reviewSummaries.length === 0) {
+      return null;
+    }
+
+    // Tally up the totals
+    let totalReviews = 0;
+    let totalPositive = 0;
+    let totalNegative = 0;
+
+    this.state.reviewSummaries.forEach((reviewSummary) => {
```
Comment (John-Michael-Long) [inline client/src/Components/ReviewsChart.jsx]: remove parens on single argument

File: client/src/components/ChartVisualization.jsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    const width = fullWidth - margins.right - margins.left;
+    const height = fullHeight - margins.top - margins.bottom;
+
+    const allDates = this.props.dataset.map(datum => datum.date);
+
+    const svg = d3.select(this.svgElement)
+      .attr('width', fullWidth)
+      .attr('height', fullHeight)
+      .append('g')
+        .attr('transform', `translate(${margins.left}, ${margins.top})`);
```
Comment (John-Michael-Long) [inline client/src/components/ChartVisualization.jsx:29]: should this be aligned with previous ?

File: client/src/components/ChartVisualization.jsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+  shouldComponentUpdate() {
+    // Prevent future-rerenders of this Component
+    return false;
+  }
+
+  render() {
+    return (
+      <div className="chart-visualization">
+        <svg ref={ (elem) => { this.svgElement = elem; } } />
```
Comment (John-Michael-Long) [inline client/src/components/ChartVisualization.jsx]: remove parens around single argument

**Review by John-Michael-Long — COMMENTED** [review]
looks good, may want to run linter for a double check

File: client/src/Components/ReviewsChart.jsx (hunk 1 above)
Comment (zmattingly) [reply client/src/Components/ReviewsChart.jsx:11]: This line was actually removed in this changeset.

File: client/src/Components/ReviewsChart.jsx (hunk 2 above)
Comment (zmattingly) [reply client/src/Components/ReviewsChart.jsx]: Done.

File: client/src/components/ChartVisualization.jsx (hunk 3 above)
Comment (zmattingly) [reply client/src/components/ChartVisualization.jsx:29]: With the way this is chained, the `.append('g')` on line 28 creates a new element. All subsequent lines are now dealing with this new element, not the previous `svg` element, so I indented line 29 to indicate this.

File: client/src/components/ChartVisualization.jsx (hunk 4 above)
Comment (zmattingly) [reply client/src/components/ChartVisualization.jsx]: Done.

**Comment by zmattingly** [comment]
I'll incorporate linter "errors" in a separate branch.

**Comment by zmattingly** [comment]
@John-Michael-Long, can you take a look at the latest changes for approval or re-assignment if there are remaining problems?

**Comment by John-Michael-Long** [comment]
looks good, merge away

**Review by John-Michael-Long — APPROVED** [review]
Approved. go a head and merge

**Pull request merged**